module/userProjects/helper.js

```diff
@@ -2172,19 +2172,86 @@ module.exports = class UserProjectsHelper {
 	 * @returns {Array} List of projects.
 	 */
 
-	static list(bodyData, pageNo, pageSize, searchText, filter) {
+	static list(pageNo, pageSize, searchText, filter, projection = []) {
 		return new Promise(async (resolve, reject) => {
 			try {
-				let projects = await projectQueries.projectDocument(
-					bodyData.query,
-					bodyData.projection,
-					bodyData.skipFields
-				)
+				let matchQuery = {
+					isDeleted: false,
+				}
+
+				//   matchQuery.status = CONSTANTS.common.ACTIVE_STATUS;
+
+				if (Object.keys(filter).length > 0) {
+					matchQuery = _.merge(matchQuery, filter)
+				}
+
+				let searchData = [
+					{
+						name: new RegExp(searchText, 'i'),
+					},
+					{
+						externalId: new RegExp(searchText, 'i'),
+					},
+					{
+						description: new RegExp(searchText, 'i'),
+					},
+				]
+
+				if (searchText !== '') {
+					if (matchQuery['$or']) {
+						matchQuery['$and'] = [{ $or: matchQuery.$or }, { $or: searchData }]
+
+						delete matchQuery.$or
+					} else {
+						matchQuery['$or'] = searchData
+					}
+				}
+
+				let projection1 = {}
```

Comment:
@Prajwal17Tunerlabs use meaningful variable name

Reply:
followed the suggestion